### PR TITLE
Rename the tree data with private identifiers

### PR DESCRIPTION
**Pull request by ttung**
### Description
Nodes on trees should be rewritten with the private identifiers that the user is privy to.

#### Issue
[ch133611](https://app.clubhouse.io/genepi/story/133611/rewrite-tree-data-with-private-identifiers)

### Test plan
Added a test with samples from own-group, can-see (private identifiers), can-see (not private identifiers), no can-see, and not even in the system.

Note that I'm stealing the `ncov_aspen.json` file for the `test_auspice_redirect.py` test; fortunately, that test is agnostic of the content of the test.

### CODE REVIEW

File: src/py/aspen/app/views/tests/data/ncov_aspen.json [hunk 1]
```diff
@@ -1,5 +1,21 @@
-[
-  {
-    "test_tree": "test_tree"
```
Comment (phoenixAja) [inline src/py/aspen/app/views/tests/data/ncov_aspen.json]: lol i was being lazy

File: src/py/aspen/app/views/tests/test_phylo_tree_rename.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+def test_phylo_tree_rename(session, mock_s3_resource, test_data_dir):
+    """Create a set of samples belonging to different groups with different levels of
+    can-see relationships.  Rename the nodes according to the can-see rules, and verify
+    that the nodes are renamed correctly."""
+    viewer_group = group_factory()
+    can_see_group = group_factory("can_see", email="can_see@yahoo.com")
+    wrong_can_see_group = group_factory(
+        "wrong_can_see", email="wrong_can_see@hotmail.com"
+    )
+    no_can_see_group = group_factory("no_can_see", email="no_can_see@aol.com")
```
Comment (phoenixAja) [inline src/py/aspen/app/views/tests/test_phylo_tree_rename.py:19]: these are some retro test email accounts lol

Comment (ttung) [reply src/py/aspen/app/views/tests/test_phylo_tree_rename.py:19]: I was trying to think what else I could think of.  juno.com?  I don't know what else is super old.

**Review by phoenixAja — APPROVED** [review]

File: src/py/aspen/app/views/phylo_trees.py [hunk 3]
```diff
@@ -84,33 +86,64 @@ def phylo_trees():
...
-            "Content-Disposition"
-        ] = f"attachment; filename={phylo_tree_id}.json"
+    response = make_response(phylo_tree_data)
+    response.headers["Content-Type"] = "text/json"
+    response.headers[
+        "Content-Disposition"
+    ] = f"attachment; filename={phylo_tree_id}.json"
 
-        return response
+    return response
```
Comment (phoenixAja) [inline src/py/aspen/app/views/phylo_trees.py]: i vote that we rename `response` -> `treesponse` here (haha jk)

**Pull request merged**